src/shader_compiler_test.cc

```diff
@@ -216,4 +220,68 @@ TEST_F(ShaderCompilerTest, ReturnsCachedShader) {
   }
 }
 
+struct ParseSpvEnvCase {
+  std::string env_str;
+  bool ok;
+  uint32_t target_env;
+  uint32_t env_version;
+  uint32_t spirv_version;
+};
+
+using ParseSpvEnvTest = ::testing::TestWithParam<ParseSpvEnvCase>;
+
+TEST_P(ParseSpvEnvTest, Samples) {
+  uint32_t target_env = 42u;
+  uint32_t env_version = 43u;
+  uint32_t spirv_version = 44u;
+  auto r = amber::ParseSpvEnv(GetParam().env_str, &target_env, &env_version,
+                              &spirv_version);
+  if (GetParam().ok) {
```

Comment:
```
if (!GetParam().ok) {
  EXPECT_FALSE(r.IsSuccess());
  return;
}
```
?

Reply:
I like mine better.  No reason to have an early return; early return muddies the cases IMHO.